docs/dev/developer-guide.md

```diff
@@ -0,0 +1,152 @@
+# `kn` Developer Guide
+
+This guide described the high-level architectural concepts of the Knative CLI client `kn`.
+It is for all developers who contribute to `kn` and want to understand some essential concepts.
+For any contributor to `kn`, it is highly recommended reading this document.
+
+_This guide is not complete and has still many gaps that we are going to fill over timer. Contributions are highly appreciated ;-)_
+
+## Flow
+
+The journey starts at [main.go]() which is the entry point when you call `kn`.
```

Comment:
This guide will be really helpful. Is the link to main file left blank to be filled in later ?

Reply:
No, not by intention, just had not the link handy but I will update it to the (future) location. There are other blank links in there, too. Thanks for the heads up, nearly forgot that ;-)